File: Java/BOJ/1522 문자열 교환.java
import java.io.*;
import java.util.*;
import java.math.*;

public class Main {
	
	static int count(String s, char x) {
		int num = 0;
		for (int i = 0; i < s.length(); i++) {
			if(s.charAt(i) == x) {
				num++;
			}
		}
		return num;
	}
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        String S = br.readLine();
        int a = count(S,'a');
        if (a > 0) {
        	S += S.substring(0,a-1);
        	int result = Integer.MAX_VALUE;
        	for (int i = 0 ; i < S.length() - (a-1); i++) {
        		int b = count(S.substring(i, i+a),'b');
        		result = Math.min(result, b);
        	}
        	System.out.println(result);        	
        }
        else {
        	System.out.println(0);
        }
    }
}

File: Python/BOJ/1522 문자열 교환.py
import sys
S = sys.stdin.readline().strip()
a = S.count('a') # a의 갯수
S += S[:a-1] # a개의 인덱스만큼 추가로 뒤에 더해주기 (초과 인덱스 고려)

result = int(10e9)
for i in range(len(S) - (a-1)) :
    result = min(result, S[i:i+a].count('b'))  # b의 갯수가 최소인 구간 찾기
print(result)